### PR TITLE
MAINT: Add nibabel as a required dependency

### PR DESCRIPTION
**Pull request by larsoner**
Just the start of a proof-of-concept for #11564, in addition to the lines I removed already in #11557. Would still need:

- [x] ~50 instances of `@requires_nibabel` to be removed
- [x] Revert `nibabel` part of #11566

### CODE REVIEW

**Comment by drammock** [comment]
I think after discussion with @agramfort @britta-wstnr @larsoner and myself today, it was agreed to abandon this idea, closing as not planned. Rationale is that adding core deps can be burdensome for users in a tightly managed environment (they may need permission from IT to get `nibabel` added, for example).

**Comment by larsoner** [comment]
Indeed I'll open another PR soon to add more `requires_nibabel` to our code

File: README.rst [hunk 1]
```diff
@@ -86,6 +86,7 @@ The minimum required dependencies to run MNE-Python are:
 - NumPy >= 1.20.2
 - SciPy >= 1.6.3
 - Matplotlib >= 3.4.0
+- NiBabel >= 3.2.1
```
Comment (jasmainak) [inline README.rst:89]: we discovered this in HNN. Looks like `nibabel` is a core dependency now but it's not reflected in the `setup.py`: https://github.com/mne-tools/mne-python/blob/main/requirements_base.txt

Comment (larsoner) [reply README.rst:89]: Can you give a traceback? We ended up *not* making it a core dependency

Comment (jasmainak) [reply README.rst:89]: here you go: https://github.com/jonescompneurolab/hnn-core/pull/649#issuecomment-1734313692

should we be installing `nibabel` in our CIs separately?

Comment (larsoner) [reply README.rst:89]: Yes, `nibabel` is an optional dependency not a core one, so you have to install it separately. PR welcome to change the wording of the error message to say something about it being optional and to consider installing it

https://github.com/mne-tools/mne-python/blob/079c868240a898204bf82b2f1bf0e04cdee75da1/mne/utils/check.py#L159

**Pull request closed without merging**